src/Storage/TSDatabaseView.m

```diff
@@ -22,40 +20,106 @@
 NSString *TSThreadDatabaseViewExtensionName  = @"TSThreadDatabaseViewExtensionName";
 NSString *TSMessageDatabaseViewExtensionName = @"TSMessageDatabaseViewExtensionName";
 NSString *TSUnreadDatabaseViewExtensionName  = @"TSUnreadDatabaseViewExtensionName";
+NSString *TSUnseenDatabaseViewExtensionName = @"TSUnseenDatabaseViewExtensionName";
+NSString *TSDynamicMessagesDatabaseViewExtensionName = @"TSDynamicMessagesDatabaseViewExtensionName";
 NSString *TSSecondaryDevicesDatabaseViewExtensionName = @"TSSecondaryDevicesDatabaseViewExtensionName";
 
 @implementation TSDatabaseView
 
-+ (BOOL)registerUnreadDatabaseView {
-    YapDatabaseView *unreadView =
-        [[TSStorageManager sharedManager].database registeredExtension:TSUnreadDatabaseViewExtensionName];
-    if (unreadView) {
++ (BOOL)registerMessageDatabaseViewWithName:(NSString *)viewName
+                               viewGrouping:(YapDatabaseViewGrouping *)viewGrouping
+                                    version:(NSString *)version
+{
+    OWSAssert(viewName.length > 0);
+    OWSAssert((viewGrouping));
+
+    YapDatabaseView *existingView = [[TSStorageManager sharedManager].database registeredExtension:viewName];
+    if (existingView) {
         return YES;
     }
 
+    YapDatabaseViewSorting *viewSorting = [self messagesSorting];
+
+    YapDatabaseViewOptions *options = [[YapDatabaseViewOptions alloc] init];
+    options.isPersistent = YES;
+    options.allowedCollections =
+        [[YapWhitelistBlacklist alloc] initWithWhitelist:[NSSet setWithObject:[TSInteraction collection]]];
+
+    YapDatabaseView *view =
+        [[YapDatabaseView alloc] initWithGrouping:viewGrouping sorting:viewSorting versionTag:version options:options];
+
+    return [[TSStorageManager sharedManager].database registerExtension:view withName:viewName];
+}
+
++ (BOOL)registerUnreadDatabaseView
+{
     YapDatabaseViewGrouping *viewGrouping = [YapDatabaseViewGrouping withObjectBlock:^NSString *(
         YapDatabaseReadTransaction *transaction, NSString *collection, NSString *key, id object) {
         if ([object conformsToProtocol:@protocol(OWSReadTracking)]) {
             id<OWSReadTracking> possiblyRead = (id<OWSReadTracking>)object;
-            if (possiblyRead.read == NO) {
+            if (!possiblyRead.wasRead && possiblyRead.shouldAffectUnreadCounts) {
                 return possiblyRead.uniqueThreadId;
             }
         }
         return nil;
     }];
 
-    YapDatabaseViewSorting *viewSorting = [self messagesSorting];
+    return [self registerMessageDatabaseViewWithName:TSUnreadDatabaseViewExtensionName
+                                        viewGrouping:viewGrouping
+                                             version:@"1"];
```

Comment:
We've changed the logic of this view, but since the only things that newly have a "read" property (info/error messages) also are already migrated as `read=YES` and they `!shouldAffectUnreadCounts` anyway so... I guess we don't have to bump the version to have clients rebuild their index since it should result in the same output.

...right?